Sources/Crypto/MAC/OTP.swift

```diff
@@ -139,7 +139,7 @@ private func generateOTP(secret: LosslessDataConvertible, algorithm: DigestAlgor
     // get last 4 bits of hash for use as offset
     let offset = Int(digest[digest.count - 1] & 0x0f)
     // get 4 bytes of the hash using offset
-    let subdigest = Data(digest[offset...offset + 3])
+    let subdigest = Data(digest[offset...offset + 3])! // Identity initializer
```

Comment:
You seem correct, but it's possibly returning a `Slice<Data>`? 
Maybe `digest.subdata(in:)` would be more appropriate?

Reply:
<img width="430" alt="screen shot 2019-01-17 at 11 47 52 am" src="https://user-images.githubusercontent.com/5561501/51344692-0da1f580-1a4e-11e9-9888-3059276c5941.png">

That's a good thought, but it doesn't look like it's a slice

Reply:
Although it's not a slice, the `Data` will have strange internal offsets. Doing `Data(...)` here performs a copy explicitly. I think you can get around this by using `data.index` instead of integer indexing, but that's a separate issue.